frontend/src/concepts/notificationPoller/NotificationPollerContext.tsx

```diff
@@ -0,0 +1,98 @@
+import * as React from 'react';
+import { AppNotification } from '~/redux/types';
+import useNotification from '~/utilities/useNotification';
+
+export type NotificationPollerContextType = {
+  watchForNotification: (item: NotificationPollerItem) => Promise<void>;
+};
+
+type NotificationPollerContextProviderProps = {
+  children: React.ReactNode;
+};
+
+export type FinalNotificationPollerResponse =
+  | ({ status: 'success' | 'error' } & Pick<AppNotification, 'title' | 'message' | 'actions'>)
+  | { status: 'stop' };
+
+export type NotificationPollerResponse = FinalNotificationPollerResponse | { status: 'repoll' };
+
+export type NotificationCallback = (signal: AbortSignal) => Promise<NotificationPollerResponse>;
+
+export type NotificationPollerItem = {
+  delayRepollMs?: number;
```

Comment:
Based on the other rename requests, maybe this?
```suggestion
  callbackDelay?: number;
```